analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EqualityOnFloatingPoint.cs

```diff
@@ -2,86 +2,132 @@
+        b = f != Single.NaN;         // Noncompliant {{Do not check floating point inequality with exact values, use 'float.IsNaN()' instead.}}
+        b = f != System.Single.NaN;  // Noncompliant {{Do not check floating point inequality with exact values, use 'float.IsNaN()' instead.}}
+    }
+}
+
+namespace TestsWithTypeAliases
+{
+    using DoubleAlias = Double;
+
+    public class ReportSpecificMessage_WithAliases
+    {
+        public void WithDoubleAlias()
         {
-            dynamic din = null;
-            if (din == null)
-            {
-            }
-
-            if (f == 3.14F) //Noncompliant {{Do not check floating point equality with exact values, use a range instead.}}
-//                ^^
-            {
-            }
-
-            if (f != 3.14F) //Noncompliant {{Do not check floating point inequality with exact values, use a range instead.}}
-            { }
-
-            if (d1 == d2) //Noncompliant
-            { }
-
-            var b = d1 == 3.14; //Noncompliant
-
-            if (true && f >= 3.146)
-            {
-            }
-
-            if (f <= 3.146 && ((f >= 3.146))) // Noncompliant indirect equality test
-            {
-            }
-            if (3.146 >= f && 3.146 <= f) // Noncompliant indirect equality test
-            {
-            }
-            if (f <= 3.146 && 3.146 <= f) // FN. Equivalent to the case above but not detected
-            {
-            }
-            if (3.146 >= f && 3.146 < f)  // Compliant no indirect equality test
-            {
-            }
-            if (f <= 3.146 && f > 3.146)  // Compliant no indirect equality test
-            {
-            }
-            var i = 3;
-            if (i <= 3 && i >= 3)
-            {
-            }
-
-            if (i < 4 || i > 4)
-            {
-            }
-
-            if (f < 3.146 || f > 3.146) // Noncompliant indirect inequality test
-            {
-            }
-            if (3.146 > f || 3.146 < f) // Noncompliant indirect inequality test
-            {
-            }
-            if (3.146 > f || f > 3.146) // FN. Equivalent to the case above but not detected
-            {
-            }
-            if (f < 3.146 || f >= 3.146) // Compliant no indirect inequality test
-            {
-            }
-            if (3.146 > f || 3.146 <= f) // Compliant no indirect inequality test
-            {
-            }
-
-            if (f <= 3.146 && true && f >= 3.146) // Not recognized
-            {
-            }
-
-            if (Math.Sign(f) == 0)
-            {
-            }
-
-            float f1 = 0.0F;
-            if ((System.Math.Sign(f1) == 0))
-            {
-            }
+            bool b;
+            DoubleAlias d1 = 1.674927471E-27;
+
+            b = d1 == DoubleAlias.NaN;    // Noncompliant {{Do not check floating point equality with exact values, use 'DoubleAlias.IsNaN()' instead.}}
```

Comment:
```suggestion
            DoubleAlias d1 = 1.674927471E-27;

            var b = d1 == DoubleAlias.NaN;    // Noncompliant {{Do not check floating point equality with exact values, use 'DoubleAlias.IsNaN()' instead.}}
```

Reply:
Actually that `bool` variable was not needed, and added nothing to the test, so I have removed it